test/lib/recordevents/observer/observer.go

```diff
@@ -82,12 +96,19 @@ func NewFromEnv(ctx context.Context, eventLogs ...recordevents.EventLog) *Observ
 		replyFunc = NoOpReply
 	}
 
-	return &Observer{
+	o := &Observer{
 		Name:      env.ObserverName,
 		EventLogs: eventLogs,
 		ctx:       ctx,
 		replyFunc: replyFunc,
 	}
+
+	if env.SkipStrategy != "" {
```

Comment:
Just a code style change, feel free to ignore it: more than having this checks, can you have a noop o.counter? so you can remove this if, always invoking `SkipperAlgorithmWithCount` and you can also remove the nil check later on too

Reply:
Done, but instead of creating nop counter, just used the linear with 0 skips which does the same thing.